crates/ruff_formatter/src/printer/mod.rs

```diff
@@ -1308,27 +1339,31 @@ impl<'a, 'print> FitsMeasurer<'a, 'print> {
             Text::Token(token) => {
                 self.state.line_width += token.len() as u32;
             }
-            Text::Text(text) => {
-                for c in text.chars() {
-                    let char_width = match c {
-                        '\t' => self.options().tab_width.value(),
-                        '\n' => {
-                            if self.must_be_flat {
-                                return Fits::No;
-                            }
-                            match args.measure_mode() {
-                                MeasureMode::FirstLine => return Fits::Yes,
-                                MeasureMode::AllLines => {
-                                    self.state.line_width = 0;
-                                    continue;
+            Text::Text { text, width } => {
+                if let Some(width) = width.width() {
```

Comment:
nit: i think i'd use a match here

Reply:
Me too but our pedantic clippy rule doesn't allow me to